### ISSUE TITLE
Kubernetes v2 deploy manifest timeout

### ISSUE THREAD

**Issue by stevegszabo** [spinnaker/spinnaker#2903]
I was able to install spinnaker using v2 manifest, pods are running however when I try to deploy a trivial service manifest as a test it hangs at: Wait For Manifest To Stabilize. The service is deployed correctly  so spinnaker is able to manage the namespace.

hal config version edit --version 1.7.6
hal deploy apply

```
    kubernetes:
      enabled: true
      accounts:
      - name: dev-kube
        requiredGroupMembership: []
        providerVersion: V2
        permissions: {}
        dockerRegistries: []
        context: devctx
        configureImagePullSecrets: true
        namespaces:
        - spinnaker
        omitNamespaces: []
        kinds:
        - spinnaker
        omitKinds: []
        customResources: []
        kubeconfigFile: /home/spinnaker/.hal/kube-config.yml
        oauthScopes: []
        oAuthScopes: []
      primaryAccount: dev-kube
```

**Comment by lwander** [comment]
What are the most recent events for the resource you're trying to deploy?

**Comment by stevegszabo** [comment]
I was using a service manifest which I deployed previously outside of spinnaker in the same namespace. I should mention that I don't have any clusters listed under the infrastructure tab. I tried adding a server group but that task failed for the same reason - Wait For Manifest To Stabilize

**Comment by stevegszabo** [comment]
I also noticed there are not options listed for Expected Artifact, I should be able to provide a bitbucket repo to pull the manifest?

**Comment by stevegszabo** [comment]
I configured a docker reg:

```
    dockerRegistry:
      enabled: true
      accounts:
      - name: nexus-docker-registry
        requiredGroupMembership: []
        providerVersion: V1
        permissions: {}
        address: https://aaaaa.com:9999
        username: aaaaaaa
        password: aaaaaaa
        email: fake.email@spinnaker.io
        cacheIntervalSeconds: 30
        clientTimeoutMillis: 60000
        cacheThreads: 1
        paginateSize: 100
        sortTagsByDate: false
        trackDigests: false
        insecureRegistry: false
        repositories: []
      primaryAccount: nexus-docker-registry

```

**Comment by stevegszabo** [comment]
I didn't configure this section which could explain why the artifacts drop down is empty:

```
  artifacts:
    gcs:
      enabled: false
      accounts: []
    github:
      enabled: false
      accounts: []
    http:
      enabled: false
      accounts: []
    s3:
      enabled: false
      accounts: []

```

**Comment by lwander** [comment]
That doesn't affect the manifest stability. Can you share the output of

```
kubect get <kind> <name> -o yaml
```

for the resource kind & name that's failing to deploy?

**Comment by stevegszabo** [comment]
```
sszabo@tor976568e1 [/home/sszabo] $ kubectl -n spinnaker get pods webapp-9c4975544-c9rcv
NAME                     READY     STATUS    RESTARTS   AGE
webapp-9c4975544-c9rcv   1/1       Running   0          39s

sszabo@tor976568e1 [/home/sszabo] $ kubectl -n spinnaker get pods webapp-9c4975544-c9rcv -o yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    artifact.spinnaker.io/location: '"spinnaker"'
    artifact.spinnaker.io/name: '"webapp"'
    artifact.spinnaker.io/type: '"kubernetes/deployment"'
    kubernetes.io/created-by: |
      {"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicaSet","namespace":"spinnaker","name":"webapp-9c4975544","uid":"a1eb041a-6ff7-11e8-bc55-6eb6c30fb769","apiVersion":"extensions","resourceVersion":"4913077"}}
    moniker.spinnaker.io/application: '"webapp"'
    moniker.spinnaker.io/cluster: '"dev-kube"'
  creationTimestamp: 2018-06-14T17:23:19Z
  generateName: webapp-9c4975544-
  labels:
    oiqclient: spinnaker
    pod-template-hash: "570531100"
  name: webapp-9c4975544-c9rcv
  namespace: spinnaker
  ownerReferences:
  - apiVersion: extensions/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: ReplicaSet
    name: webapp-9c4975544
    uid: a1eb041a-6ff7-11e8-bc55-6eb6c30fb769
  resourceVersion: "4913094"
  selfLink: /api/v1/namespaces/spinnaker/pods/webapp-9c4975544-c9rcv
  uid: a1ebac66-6ff7-11e8-bc55-6eb6c30fb769
spec:
  containers:
  - image: xxxxxxxxxxxx:9999/spinnaker/spinnaker:84
    imagePullPolicy: IfNotPresent
    name: webapp
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-tb5kc
      readOnly: true
  dnsPolicy: ClusterFirst
  imagePullSecrets:
  - name: spinnaker-nexus
  nodeName: XXXXXXXXXXXXXXXXX
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: default
  serviceAccountName: default
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.alpha.kubernetes.io/notReady
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.alpha.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - name: default-token-tb5kc
    secret:
      defaultMode: 420
      secretName: default-token-tb5kc
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: 2018-06-14T17:23:19Z
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: 2018-06-14T17:23:22Z
    status: "True"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: 2018-06-14T17:23:19Z
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: docker://395aee0ede89f9259e591de5db5f867311f44a6934a17a3834211aef6d5a0764
    image: xxxxxx.com:9999/spinnaker/spinnaker:84
    imageID: docker-pullable://xxxxxx.com:9999/spinnaker/spinnaker@sha256:a3c833825d07dfb056a0423f3c3f7ee8176a19b5e929cd5c0da6208a30a6307c
    lastState: {}
    name: webapp
    ready: true
    restartCount: 0
    state:
      running:
        startedAt: 2018-06-14T17:23:22Z
  hostIP: 10.166.221.224
  phase: Running
  podIP: XXXXXXXXXXXXXXXXX
  qosClass: BestEffort
  startTime: 2018-06-14T17:23:19

```

**Comment by lwander** [comment]
Were you deploying the pod, or a deployment managing the pod? If it's the
latter, can you share the same output but for the deployment?

**Comment by stevegszabo** [comment]
Here is the deployment:

```
sszabo@tor976568e1 [/home/sszabo] $ kubectl -n spinnaker get -o yaml deployment webapp
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "13"
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"apps/v1beta1","kind":"Deployment","metadata":{"annotations":{},"labels":{"oiqclient":"spinnaker"},"name":"webapp","namespace":"spinnaker"},"spec":{"minReadySeconds":60,"progressDeadlineSeconds":120,"replicas":1,"selector":{"matchLabels":{"oiqclient":"spinnaker"}},"strategy":{"rollingUpdate":{"maxSurge":10,"maxUnavailable":1},"type":"RollingUpdate"},"template":{"metadata":{"labels":{"oiqclient":"spinnaker"},"name":"webapp"},"spec":{"containers":[{"image":"aaaaaa.com:9999/spinnaker/spinnaker:106","name":"webapp"}],"imagePullSecrets":[{"name":"spinnaker-nexus"}]}}}}
  creationTimestamp: 2018-06-14T18:36:22Z
  generation: 13
  labels:
    oiqclient: spinnaker
  name: webapp
  namespace: spinnaker
  resourceVersion: "4916553"
  selfLink: /apis/extensions/v1beta1/namespaces/spinnaker/deployments/webapp
  uid: d65b7ec3-7001-11e8-bc55-6eb6c30fb769
spec:
  minReadySeconds: 60
  progressDeadlineSeconds: 120
  replicas: 1
  revisionHistoryLimit: 2
  selector:
    matchLabels:
      oiqclient: spinnaker
  strategy:
    rollingUpdate:
      maxSurge: 10
      maxUnavailable: 1
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        oiqclient: spinnaker
      name: webapp
    spec:
      containers:
      - image: aaaaaa.com:9999/spinnaker/spinnaker:106
        imagePullPolicy: IfNotPresent
        name: webapp
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
      dnsPolicy: ClusterFirst
      imagePullSecrets:
      - name: spinnaker-nexus
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
status:
  availableReplicas: 1
  conditions:
  - lastTransitionTime: 2018-06-14T18:36:22Z
    lastUpdateTime: 2018-06-14T18:36:22Z
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
  - lastTransitionTime: 2018-06-14T18:36:22Z
    lastUpdateTime: 2018-06-14T19:15:40Z
    message: ReplicaSet "webapp-7b654f5c5f" has successfully progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  observedGeneration: 13
  readyReplicas: 1
  replicas: 1
  updatedReplicas: 1

```

**Comment by stevegszabo** [comment]
I found this in the orca log:

2018-06-18 15:07:43.902  WARN 1 --- [    handlers-19] c.n.s.o.c.t.m.WaitForManifestStableTask  : [anonymous] Unable to read manifest 'ingress webapp' in 'spinnaker' for account dev-kube
retrofit.RetrofitError: 404 

2018-06-18 15:07:48.907 ERROR 1 --- [    handlers-19] c.n.s.orca.q.handler.RunTaskHandler      : [] Error running WaitForManifestStableTask for pipeline[87357ecf-61fa-41ad-9674-912c61f56c15]
com.netflix.spinnaker.orca.exceptions.TimeoutException: Stage Deploy timed out after 1 minute

**Comment by lwander** [comment]
Thanks @stevegszabo -- did you set the timeout to 1 minute? The default for that stage is 30 minutes IIRC.

**Comment by stevegszabo** [comment]
Yes I set to 1 min, I see these in the gate log:

com.netflix.spinnaker.gate.services.ManifestService$ManifestNotFound: Unable to find Ingress webapp in dev-kube/spinnaker
com.netflix.spinnaker.gate.services.ManifestService$ManifestNotFound: Unable to find Deployment webapp in dev-kube/spinnaker
com.netflix.spinnaker.gate.services.ManifestService$ManifestNotFound: Unable to find Service webapp in dev-kube/spinnaker

**Comment by lwander** [comment]
Increase the time from 1 minute -- the cache is probably in the process of updating.

**Comment by lwander** [comment]
Try the default timeout. If you see 404s all the way until the end, then something else is wrong.

**Comment by stevegszabo** [comment]
From orca the clouddriver doesn't seem to have any manifests:

```
bash-4.4$ wget http://spin-clouddriver.spinnaker:7002/ && cat index.html
Connecting to spin-clouddriver.spinnaker:7002 (XXXXXXXXXXXX:7002)
index.html           100% |*************************************************************************************************|   110   0:00:00 ETA
{
  "_links" : {
    "profile" : {
      "href" : "http://spin-clouddriver.spinnaker:7002/profile"
    }
  }
}

```

**Comment by stevegszabo** [comment]
Same results using default timeout of 30 mins

**Comment by lwander** [comment]
> bash-4.4$ wget http://spin-clouddriver.spinnaker:7002/ && cat index.html

Not sure what the goal of that is -- there is a separate /manifest controller.

Are there any errors in the logs for clouddriver? Does the ingress "webapp" exist in your cluster? My guess is something is failing to be cached & the logs might show us what's going wrong.

**Comment by stevegszabo** [comment]
Orca is trying to access the manifest which doesn't exist:

HTTP 404 http://spin-clouddriver.spinnaker:7002/manifests/dev-kube/spinnaker/ingress%20webapp

The deployment is successful, the resources in the manifest are created in the correct namespace:

Deployment
Service
Ingress

**Comment by stevegszabo** [comment]
```
2018-06-18 16:17:07.030  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [ORCHESTRATION] - Processing op: KubernetesDeployManifestOperation
2018-06-18 16:17:07.032  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Beginning deployment of manifest...
2018-06-18 16:17:07.033  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Finding deployer for service...
2018-06-18 16:17:07.034  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Swapping out artifacts in service webapp from context...
2018-06-18 16:17:07.035  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Finding deployer for ingress...
2018-06-18 16:17:07.037  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Swapping out artifacts in ingress webapp from context...
2018-06-18 16:17:07.039  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Finding deployer for deployment...
2018-06-18 16:17:07.040  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Swapping out artifacts in deployment webapp from context...
2018-06-18 16:17:07.041  INFO 1 --- [ool-4-thread-10] c.n.s.c.k.v.a.ArtifactReplacer$Replacer  : Found valid swap for Artifact(type=docker/image, name=aaaaaaaaaaa.com:9999/spinnaker/spinnaker, version=100, location=null, reference=aaaaaaaaaaa.com:9999/spinnaker/spinnaker:100, metadata=null, artifactAccount=null, provenance=null, uuid=null) using $.spec.template.spec.containers.[?( @.image == "aaaaaaaaaaa.com:9999/spinnaker/spinnaker" )].image: ["aaaaaaaaaaa.com:9999/spinnaker/spinnaker"]
2018-06-18 16:17:07.043  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Checking if all requested artifacts were bound...
2018-06-18 16:17:07.044  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Sorting manifests by priority...
2018-06-18 16:17:07.045  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Finding deployer for ingress...
2018-06-18 16:17:07.047  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Finding deployer for service...
2018-06-18 16:17:07.048  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Finding deployer for deployment...
2018-06-18 16:17:07.050  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Finding deployer for ingress...
2018-06-18 16:17:07.051  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Deploy order is: service webapp, ingress webapp, deployment webapp
2018-06-18 16:17:07.052  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Finding deployer for service...
2018-06-18 16:17:07.054  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Annotating manifest service webapp with artifact, relationships & moniker...
2018-06-18 16:17:07.055  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Swapping out artifacts in service webapp from other deployments...
2018-06-18 16:17:07.057  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Submitting manifest service webapp to kubernetes master...
2018-06-18 16:17:07.760  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Finding deployer for ingress...
2018-06-18 16:17:07.762  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Annotating manifest ingress webapp with artifact, relationships & moniker...
2018-06-18 16:17:07.763  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Swapping out artifacts in ingress webapp from other deployments...
2018-06-18 16:17:07.765  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Submitting manifest ingress webapp to kubernetes master...
2018-06-18 16:17:08.468  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Finding deployer for deployment...
2018-06-18 16:17:08.469  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Annotating manifest deployment webapp with artifact, relationships & moniker...
2018-06-18 16:17:08.470  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Swapping out artifacts in deployment webapp from other deployments...
2018-06-18 16:17:08.472  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [DEPLOY_KUBERNETES_MANIFEST] - Submitting manifest deployment webapp to kubernetes master...
2018-06-18 16:17:09.114  INFO 1 --- [roviderConfig-1] c.n.s.c.k.v.c.KubernetesV2ProviderConfig : Adding 0 agents for account dev-kube
2018-06-18 16:17:09.176  INFO 1 --- [ool-4-thread-10] c.n.s.c.data.task.jedis.JedisTask        : [ORCHESTRATION] - Orchestration completed.

```

**Comment by lwander** [comment]
We'll need more than just the deployment logs, but a few minute window of caching activity.

**Comment by stevegszabo** [comment]
Is this relevant? [Adding 0 agents](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/KubernetesV2ProviderConfig.java#L112)?

2018-06-18 20:48:39.114  INFO 1 --- [roviderConfig-1] c.n.s.c.k.v.c.KubernetesV2ProviderConfig : Adding 0 agents for account tordevkubecluster01

**Comment by stevegszabo** [comment]
I see entries in relation to the docker reg but nothing else for k8s:

```
2018-06-18 20:54:35.822  INFO 1 --- [cutionAction-11] .d.r.p.a.DockerRegistryImageCachingAgent : Describing items in dev-nexus-docker-registry/DockerRegistryImageCachingAgent[1/1]
2018-06-18 20:54:35.826  INFO 1 --- [cutionAction-11] .d.r.p.a.DockerRegistryImageCachingAgent : Caching 341 tagged images in dev-nexus-docker-registry/DockerRegistryImageCachingAgent[1/1]
2018-06-18 20:54:35.826  INFO 1 --- [cutionAction-11] .d.r.p.a.DockerRegistryImageCachingAgent : Caching 341 image ids in dev-nexus-docker-registry/DockerRegistryImageCachingAgent[1/1]

```

**Comment by stevegszabo** [comment]
Is there a way to increase the log level for a v2 deployment?

**Comment by lwander** [comment]
Are there no exceptions anywhere near the top of the logs? 

```
logging.level.root: DEBUG
```

in `~/.hal/default/profiles/clouddriver-local.yml` will help

**Comment by lwander** [comment]
Yup that's right, it's only for custom config.

**Comment by stevegszabo** [comment]
running in DEBUG mode:

```
2018-06-20 00:08:03.724 DEBUG 1 --- [0.0-7002-exec-6] w.c.HttpSessionSecurityContextRepository : SecurityContext is empty or contents are anonymous - context will not be stored in HttpSession.
2018-06-20 00:08:03.724 DEBUG 1 --- [0.0-7002-exec-6] s.s.w.c.SecurityContextPersistenceFilter : SecurityContextHolder now cleared, as request processing completed
2018-06-20 00:08:03.724 DEBUG 1 --- [0.0-7002-exec-6] o.s.b.w.f.OrderedRequestContextFilter    : Cleared thread-bound request context: org.apache.catalina.connector.RequestFacade@5d749631
2018-06-20 00:08:03.725 DEBUG 1 --- [0.0-7002-exec-6] o.a.tomcat.util.net.SocketWrapperBase    : Socket: [org.apache.tomcat.util.net.NioEndpoint$NioSocketWrapper@209cd54c:org.apache.tomcat.util.net.NioChannel@43ce05e1:java.nio.channels.SocketChannel[connected local=/127.0.0.1:7002 remote=/127.0.0.1:50570]], Read from buffer: [0]
2018-06-20 00:08:03.725 DEBUG 1 --- [0.0-7002-exec-6] o.apache.coyote.http11.Http11Processor   : Socket: [org.apache.tomcat.util.net.NioEndpoint$NioSocketWrapper@209cd54c:org.apache.tomcat.util.net.NioChannel@43ce05e1:java.nio.channels.SocketChannel[connected local=/127.0.0.1:7002 remote=/127.0.0.1:50570]], Status in: [OPEN_READ], State out: [OPEN]
2018-06-20 00:08:03.725 DEBUG 1 --- [0.0-7002-exec-7] o.a.tomcat.util.net.SocketWrapperBase    : Socket: [org.apache.tomcat.util.net.NioEndpoint$NioSocketWrapper@209cd54c:org.apache.tomcat.util.net.NioChannel@43ce05e1:java.nio.channels.SocketChannel[connected local=/127.0.0.1:7002 remote=/127.0.0.1:50570]], Read from buffer: [0]
2018-06-20 00:08:03.726 DEBUG 1 --- [0.0-7002-exec-7] o.apache.coyote.http11.Http11Processor   : Error parsing HTTP request header

java.io.EOFException: null
	at org.apache.tomcat.util.net.NioEndpoint$NioSocketWrapper.fillReadBuffer(NioEndpoint.java:1244) ~[tomcat-embed-core-8.5.20.jar:8.5.20]
	at org.apache.tomcat.util.net.NioEndpoint$NioSocketWrapper.read(NioEndpoint.java:1184) ~[tomcat-embed-core-8.5.20.jar:8.5.20]
	at org.apache.coyote.http11.Http11InputBuffer.fill(Http11InputBuffer.java:713) ~[tomcat-embed-core-8.5.20.jar:8.5.20]
	at org.apache.coyote.http11.Http11InputBuffer.parseRequestLine(Http11InputBuffer.java:362) ~[tomcat-embed-core-8.5.20.jar:8.5.20]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:683) ~[tomcat-embed-core-8.5.20.jar:8.5.20]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66) [tomcat-embed-core-8.5.20.jar:8.5.20]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:868) [tomcat-embed-core-8.5.20.jar:8.5.20]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1457) [tomcat-embed-core-8.5.20.jar:8.5.20]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-embed-core-8.5.20.jar:8.5.20]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_151]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_151]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-embed-core-8.5.20.jar:8.5.20]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_151]

2018-06-20 00:08:03.726 DEBUG 1 --- [0.0-7002-exec-7] o.apache.coyote.http11.Http11Processor   : Socket: [org.apache.tomcat.util.net.NioEndpoint$NioSocketWrapper@209cd54c:org.apache.tomcat.util.net.NioChannel@43ce05e1:java.nio.channels.SocketChannel[connected local=/127.0.0.1:7002 remote=/127.0.0.1:50570]], Status in: [OPEN_READ], State out: [CLOSED]
2018-06-20 00:08:03.726 DEBUG 1 --- [0.0-7002-exec-7] o.apache.tomcat.util.threads.LimitLatch  : Counting down[http-nio-0.0.0.0-7002-exec-7] latch=1
2018-06-20 00:08:05.069 DEBUG 1 --- [roviderConfig-1] o.s.b.f.s.DefaultListableBeanFactory     : Creating instance of bean 'synchronizeKubernetesV1Provider'
2018-06-20 00:08:05.069 DEBUG 1 --- [roviderConfig-1] o.s.b.f.s.DefaultListableBeanFactory     : Returning cached instance of singleton bean 'kubernetesV1ProviderConfig'
2018-06-20 00:08:05.071 DEBUG 1 --- [roviderConfig-1] o.s.b.f.s.DefaultListableBeanFactory     : Finished creating instance of bean 'synchronizeKubernetesV1Provider'
2018-06-20 00:08:05.317 DEBUG 1 --- [roviderConfig-1] o.s.b.f.s.DefaultListableBeanFactory     : Creating instance of bean 'synchronizeKubernetesV2Provider'
2018-06-20 00:08:05.317 DEBUG 1 --- [roviderConfig-1] o.s.b.f.s.DefaultListableBeanFactory     : Returning cached instance of singleton bean 'kubernetesV2ProviderConfig'
2018-06-20 00:08:05.318  INFO 1 --- [roviderConfig-1] c.n.s.c.k.v.c.KubernetesV2ProviderConfig : Adding 0 agents for account tordevkubecluster01
```

**Comment by lwander** [comment]
It's possible our nightly integration tests just caught the same bug -- can run 

`grep 'kinds' -C 3 ~/.hal/config`?

**Comment by lwander** [comment]
Ah great -- can you (using a text editor rather than halyard) set `kinds: []` in ~/.hal/config? And then `hal deploy apply`

**Comment by stevegszabo** [comment]
After applying the above I now see a k8s cluster listed under the cluster tab!

**Comment by lwander** [comment]
Nice! Halyard 1.3.1 has the fix that prevents `--namespaces` from editing the `kinds: ` field, so in the future this shouldn't happen.

**Comment by lwander** [comment]
Let me know if the deploys succeed & we'll close this issue.

**Comment by stevegszabo** [comment]
Deployment was successful, thank you!

**Comment by dklostermann** [comment]
Sorry to reopen this after you just closed it, but I'm experiencing the same problem.  My Service, Ingress, and deployment all deploy with no issues and I can get data from this new deployment, but Spinnaker is reporting it as failed due to time out:

Exception ( Force Cache Refresh )
timeout

I ran grep 'kinds' -C 3 ~/.hal/config to check the kinds and this is the result:

        configureImagePullSecrets: true
        namespaces: []
        omitNamespaces: []
        kinds: []
        omitKinds: []
        customResources: []

Is there something else to check?

**Comment by lwander** [comment]
What you're seeing is https://github.com/spinnaker/spinnaker/issues/2494

It's resolved in 1.8 (the candidate can be tried with `hal config version edit --version release-1.8.x-latest-unvalidated`).
